libqtile/group.py

```diff
@@ -201,12 +204,14 @@ def add(self, win, focus=True):
 
     def remove(self, win):
         self.windows.remove(win)
+        currentWindow = win.group.currentWindow
         win.group = None
         nextfocus = None
         if win.floating:
             nextfocus = self.floating_layout.remove(win)
             if nextfocus is None:
-                nextfocus = self.layout.focus_first()
+                nextfocus = self.layout.focus_first() \
+                    if self.is_layout_floating() else currentWindow
```

Comment:
This seems like a bit more of a general case: if the currentWindow is not win, we don't want to mess with the focus at all; should we just return after removing it above if that's the case?